drivers/gpio/gpio_ite_it8801.c

```diff
@@ -0,0 +1,486 @@
+		if (control & IT8801_GPIOIOT_OD) {
+			flags |= GPIO_OPEN_DRAIN;
+		}
+
+		ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_ipsr, &value);
+		if (ret) {
+			LOG_ERR("Failed to read pin status (ret %d)", ret);
+			return ret;
+		}
+
+		/* Get GPIO output level */
+		if (value & mask) {
+			flags |= GPIO_OUTPUT_HIGH;
+		} else {
+			flags |= GPIO_OUTPUT_LOW;
+		}
+	} else {
+		flags |= GPIO_INPUT;
+	}
+
+	/* pullup / pulldown */
+	if (control & IT8801_GPIOPUE) {
+		flags |= GPIO_PULL_UP;
+	} else if (control & IT8801_GPIOPDE) {
+		flags |= GPIO_PULL_DOWN;
+	}
+
+	*out_flags = flags;
+
+	return 0;
+}
+#endif
+
+static int gpio_it8801_port_get_raw(const struct device *dev, gpio_port_value_t *value)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+	uint8_t val;
+
+	/* Get raw bits of GPIO mirror register */
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_ipsr, &val);
+	if (ret) {
+		LOG_ERR("Failed to get port mask (ret %d)", ret);
+		return ret;
+	}
+
+	*value = val;
+
+	return ret;
+}
+
+static int gpio_it8801_port_set_masked_raw(const struct device *dev, gpio_port_pins_t mask,
+					   gpio_port_value_t value)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+
+	ret = i2c_reg_update_byte_dt(&config->i2c_dev, config->reg_sovr, mask, value);
+	if (ret) {
+		LOG_ERR("Failed to set port mask (ret %d)", ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_port_set_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+
+	/* Set raw bits of GPIO data register */
+	ret = i2c_reg_update_byte_dt(&config->i2c_dev, config->reg_sovr, pins, pins);
+	if (ret) {
+		LOG_ERR("Failed to set bits raw (ret %d)", ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_port_clear_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+
+	/* Clear raw bits of GPIO data register */
+	ret = i2c_reg_update_byte_dt(&config->i2c_dev, config->reg_sovr, pins, 0);
+	if (ret) {
+		LOG_ERR("Failed to clear bits raw (ret %d)", ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_port_toggle_bits(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+	uint8_t val, new_val;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_sovr, &val);
+	if (ret) {
+		return ret;
+	}
+	/* Toggle raw bits of GPIO data register */
+	new_val = val ^ pins;
+	if (new_val != val) {
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_sovr, new_val);
+		if (ret) {
+			LOG_ERR("Failed to write toggle value (ret %d)", ret);
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_manage_callback(const struct device *dev, struct gpio_callback *callback,
+				       bool set)
+{
+	struct gpio_it8801_data *data = dev->data;
+	int rc;
```

Comment:
`int ret`, for coherency with the rest of the file

Reply:
Done.